virttest/test_setup.py

```diff
@@ -316,6 +329,40 @@ def __init__(self, params):
 
         self.target_hugepages = target_hugepages
 
+    def check_1G_hugepage_support(self):
+        """
+        Check whether the host support 1G hugepage.
+        The flag in cpuinfo should be "pdpe1gb".
+        """
+        error_context.context("Check whether the host support 1G hugepage")
+        cpuinfo = file('/proc/cpuinfo').read()
+        cpu_flag = "pdpe1gb"
+        if cpu_flag not in cpuinfo:
+            raise HPNotSupportedError("Host does not support 1G support,"
+                                      "no such %s flag in host." % cpu_flag)
+        self.check_1G_hugepage_kernel_cmdline()
+
+    def check_1G_hugepage_kernel_cmdline(self):
+        """
+        Check whether host kernel cmdline support 1G hugepage support.
+        Needs option "hugepagesz=1G" in kernel cmdline.
+        """
+        error_context.context("Check host kernel cmdline support 1G hugepage")
+        kernel_cmdline = file('/proc/cmdline').read()
+        if "hugepagesz=1G" not in kernel_cmdline:
+            raise HPSetupError("Please check whether the host kernel"
+                               "cmdline has hugepagesz=1G option\n"
+                               "Host kernel cmdline is %s" % kernel_cmdline)
+
```

Comment:
BTW, another way to implement `check_1G_hugepage_kernel_cmdline` is abstract it as 'check_kernel_cmdline(pattern)` into utils_misc.py, then check whatever pattern(such as 'hugepage=1G') you want.